File: app/src/main/java/com/savvasdalkitsis/gameframe/ip/repository/IpRepository.java
package com.savvasdalkitsis.gameframe.ip.repository;

import com.savvasdalkitsis.gameframe.ip.model.IpAddress;

import rx.Observable;

public interface IpRepository {

    Observable<IpAddress> getIpAddress();

    void saveIpAddress(IpAddress ipAddress);
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/ip/repository/IpRepository.kt
package com.savvasdalkitsis.gameframe.ip.repository

import com.savvasdalkitsis.gameframe.ip.model.IpAddress
import io.reactivex.Single

interface IpRepository {

    val ipAddress: Single<IpAddress>

    fun saveIpAddress(ipAddress: IpAddress)
}
